src/java.desktop/macosx/classes/com/apple/laf/AquaButtonUI.java

```diff
@@ -305,7 +305,15 @@ public void paint(final Graphics g, final JComponent c) {
         }
 
         // performs icon and text rect calculations
-        final String text = layoutAndGetText(g, b, aquaBorder, i, viewRect, iconRect, textRect);
+        final String text;
+        final View v = (View)c.getClientProperty(BasicHTML.propertyKey);
+        if (v != null) {
+            // use zero insets for view since layout only handles text calculations
+            text = layoutAndGetText(g, b, aquaBorder, new
+                    Insets(0,0,0,0), viewRect, iconRect, textRect);
```

Comment:
If there is only html text instead of image in JButton, will it not be a problem using 0 insets or is there a problem with text too being shifted like image?

Reply:
The text was also unexpectedly shifted, so this zero inset fixes that issue as well. Setting the insets to zero for all HTML fixes the unwanted padding.

Reply:
Should we create a separate test for HTML text? Probably manual only but still. If it is a known issue and it's also resolved by this fix, I find such a test useful.

Reply:
I can definitely try to do so. The newest version of the test is now headless using BufferedImage. By manual, do you mean making a headful test where the tester clicks pass or fail according to the position of HTML text on the button?